File: IsraelHiking.Web/android/app/src/main/java/com/mapeak/car/CarMarkerData.kt
package com.mapeak.car

import org.json.JSONException
import org.json.JSONObject
import org.maplibre.android.geometry.LatLng

/** A private route point (POI), as fed from JS: `{ latlng: [lng, lat], title }`. */
data class CarMarkerData(
        val lngLat: LatLng,
        val title: String = ""
) {
    companion object {
        @Throws(JSONException::class)
        fun fromJson(json: JSONObject): CarMarkerData {
            val latlng = json.getJSONArray("latlng")
            return CarMarkerData(
                    lngLat = LatLng(latlng.getDouble(1), latlng.getDouble(0)),
                    title = json.optString("title")
            )
        }
    }
}

File: IsraelHiking.Web/ios/App/App/Car/CarMarkerData.swift
import CoreLocation
import Foundation

/// A private route point (POI), mirroring `CarMarkerData.kt`: `{ latlng: [lng, lat], title }`.
struct CarMarkerData {
    let coordinate: CLLocationCoordinate2D
    let title: String

    static func from(_ json: [String: Any]) -> CarMarkerData? {
        guard let pair = json["latlng"] as? [Any], pair.count >= 2,
              let lng = (pair[0] as? NSNumber)?.doubleValue,
              let lat = (pair[1] as? NSNumber)?.doubleValue
        else { return nil }
        return CarMarkerData(
            coordinate: CLLocationCoordinate2D(latitude: lat, longitude: lng),
            title: (json["title"] as? String) ?? ""
        )
    }
}
